jerry-ext/include/jerryscript-ext/args.h

```diff
@@ -0,0 +1,115 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_ARGS_H
+#define JERRYX_ARGS_H
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "jerryscript.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* __cplusplus */
+
+typedef struct jerryx_arg_t jerryx_arg_t;
+typedef struct jerryx_arg_iterator_t jerryx_arg_iterator_t;
+
+/**
+ * Signature of the transform funtion.
+ */
+typedef jerry_value_t (*jerryx_arg_transform_func_t) (jerryx_arg_iterator_t *iterator_p);
```

Comment:
this is very personal taste (and signature will change anyway), but the name of the function parameter should not be simply `iterator_p`, which just repeats the name of the type of the parameter, but should also tell something about the data (e.g., in the previously mentioned commits, js_args was used to highlight that it's an iterator over the JS arguments -- not necessarily the best name, could be js_arg_it, or ...)